authorization/overview.mdx

```diff
@@ -0,0 +1,276 @@
+- `input.request`: The incoming request details, such as the `namespace`, `resource`, and `verb`.
+
+### Authentication Information
+
+Flipt provides the raw authentication information to OPA for evaluation. This information is specific to the authentication method used to authenticate the request.
+
+For example, if you're using the [OIDC authentication method](/authentication/methods#openid-connect), the `input.authentication.metadata` field may contain the user's name and email as well as custom claims assigned to the user.
+
+Here is an example of the `input.authentication.metadata` field for a request authenticated using a example OIDC provider:
+
+```json
+{
+    "io.flipt.auth.email": "user@email.com",
+    "io.flipt.auth.name": "John Doe"
+    "io.flipt.auth.claims": {
+        "roles": ["admin", "viewer"]
+    }
+}
+```
+
+The following fields are available in the `input.authentication` field:
+
+- `metadata`: A map of authentication metadata provided by the authentication method. This can include the user's email, name, roles, etc.
+  - `io.flipt.auth.email`: The user's email address.
+  - `io.flipt.auth.name`: The user's name.
+  - `io.flipt.auth.claims`: A map of **all** claims provided by the authentication method. This can include the user's roles, groups, etc. These claims are marshaled into a JSON string before being passed to OPA for evaluation.
+
+#### Mapping User Information
+
+Each authentication method used to authenticate requests with Flipt will provide different information about the user. It's up to you to map this information to roles that can be used in your policies.
+
+Some authentication methods provide user details such as roles directly, while others may provide a user ID or email that you can use to look up roles in your own system. Many authentication providers support adding custom claims to the JWT token, which can be used to provide additional information about the user.
+
+For example, [Okta](https://www.okta.com/) allows you to add custom claims using their groups feature. An example JWT token with custom claims generated by Okta might look like this:
+
+```json
+{
+  "sub": "00uixa271s6x7qt8I0h7",
+  "ver": 1,
+  "iss": "https://{yourOktaDomain}",
+  "aud": "0oaoiuhhch8VRtBnC0h7",
+  "iat": 1574201516,
+  "exp": 1574205116,
+  "jti": "ID.ewMNfSvcpuqyS93OgVeCN3F2LseqROkyYjz7DNb9yhs",
+  "amr": ["pwd", "mfa", "kba"],
+  "idp": "00oixa26ycdNcX0VT0h7",
+  "nonce": "UBGW",
+  "auth_time": 1574201433,
+  "groups": ["Everyone", "IT"]
+}
+```
+
+In this example, the `groups` claim is used to provide the user's organizational groups. You can then write a policy that checks for the presence of specific groups to determine whether the user should be allowed to access a particular resource.
+
+```rego policy.rego
+package flipt.authz.v1
+
+default allow = false
+
+allow {
+    claims := json.unmarshal(input.authentication.metadata["io.flipt.auth.claims"])
```

Comment:
We can drop the unmarshal now right?
```suggestion
    claims := input.authentication.metadata["io.flipt.auth.claims"]
```

Reply:
the requirement to `json.unmarshal` is still there. I think we could potentially unmarshal it for them in the future and add to another field on the authentication, but for now I left it as is